certlint_test.go

```diff
@@ -46,37 +51,166 @@ accZ5e1iOtNIBTVUEaQ0YxNwhZqnQO4rgBFCnEpNH6TB62hXUD6u/oP/WB48Wmek
 KPgaoWHMo5sOFQnw5A==
 -----END CERTIFICATE-----`
 
+func TestMain(m *testing.M) {
+	build := exec.Command("go", "build")
+	berr := build.Run()
+	if berr != nil {
+		fmt.Printf("could not build before tests: %v", berr)
+		os.Exit(1)
+	}
+
+	retcode := m.Run()
+
+	clean := exec.Command("go", "clean", "-x")
+	cerr := clean.Run()
+	if cerr != nil {
+		fmt.Printf("could not cleanup after tests: %v", cerr)
+		os.Exit(1)
+	}
+
+	os.Exit(retcode)
+}
+
 func TestTestData(t *testing.T) {
 	var icaCache = lru.New(200)
 
 	// TODO: Check for specific errors per certificate to be sure we don't miss one
-	files, _ := ioutil.ReadDir("./testdata")
+	files, _ := filepath.Glob("./testdata/*.pem")
 	for _, f := range files {
-		fmt.Printf("---- %s ----\n", f.Name())
+		fname := path.Base(f)
+		fmt.Printf("---- %s ----\n", fname)
 
-		der := getCertificate("./testdata/" + f.Name())
+		der := getCertificate("./testdata/" + fname)
 		if len(der) > 0 {
 			result := do(icaCache, der, true, true)
 			if len(result.Errors.List()) == 0 {
-				t.Errorf("Expected some errors, got %d in %s", len(result.Errors.List()), f.Name())
+				t.Errorf("Expected some errors, got %d in %s", len(result.Errors.List()), fname)
 				continue
 			}
 			for _, err := range result.Errors.List() {
-				fmt.Printf("%s (%s)\n", err.Error(), result.Type)
+				fmt.Printf("Priority: %s, Message: %s (%s)\n", err.Priority(), err.Error(), result.Type)
 			}
 		}
 	}
 }
 
+func TestCLITestData(t *testing.T) {
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	testdata := path.Join(dir, "testdata")
+	files, _ := filepath.Glob("./testdata/*.pem")
+	for _, f := range files {
```

Comment:
Is it relevant to run the test against all test data again? The test data is a resource that should be growing to cover much more cases to make sure we detect and will always detect them, running these test twice looks to be a waste of resources and time.

Should we limit the test to a few relevant checks to just check if the CLI does what it should be doing?

Reply:
Yep, that approach sounds fine to me.